modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts

```diff
@@ -24,15 +24,18 @@ import * as InsertList from './InsertList';
 
 const wrappedElements = [ 'pre' ];
 
-const shouldPasteContentOnly = (fragment: AstNode, parentNode: Element) => {
-  const firstNode = fragment.firstChild;
-
-  const isAFlattenableTag = Arr.contains(wrappedElements, firstNode.name);
-  const isPastingInTheSameTag = firstNode.name === parentNode.tagName.toLowerCase();
-  const lastNode = fragment.lastChild.attr('data-mce-type') === 'bookmark' ? fragment.lastChild.prev : fragment.lastChild;
-  const isCopingOnlyOneTag = firstNode === lastNode;
-
-  return isCopingOnlyOneTag && isAFlattenableTag && isPastingInTheSameTag;
+const shouldPasteContentOnly = (dom: DOMUtils, fragment: AstNode, parentNode: Element, editor: Editor): boolean => {
+  const firstNode = fragment.firstChild as AstNode;
+  const lastNode = fragment.lastChild as AstNode;
+  const last = lastNode.attr('data-mce-type') === 'bookmark' ? lastNode.prev : lastNode;
+
+  const isCopyingOnlyOneTag = firstNode === last;
```

Comment:
We're also getting a lot of checks in this function now that eagerly evaluate. Most are quick but some are slower... I'd suggest breaking this up to something like:
```
const isCopyingOnlyOneTag = firstNode === last;
const isWrappedElement = Arr.contains(wrappedElements, firstNode.name);
if (isCopyingOnlyOneTag && isWrappedElement) {
  ...other checks
} else {
  return false;
}
```